### PR TITLE
Change starter template to make frontend external endpoints

### LINKED ISSUE #3495: Consider changing templates to use WithExternalHttpEndpoints
With the changes, out of the box a deployed Aspire starter app to Azure (or other place) is not marking the front-end accessible.

Consider adding `WithExternalHttpEndpoints` in the templates for the endpoints that need it (front-end, Aspire Starter)

### PR DESCRIPTION
**Pull request by timheuer**
Fixes #3495 to make the frontend web app in aspire-starter template to use the `WithExternalHttpEndpoints` 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3514)

### CODE REVIEW

File: src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #if UseRedisCache
     .WithReference(cache)
 #endif
-    .WithReference(apiService);
+    .WithReference(apiService)
+    .WithExternalHttpEndpoints();
```
Comment (davidfowl) [inline src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs:14]: I would add it first not last.

Comment (timheuer) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs:14]: Curious why? is there some race?

Comment (davidfowl) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs:14]: Not a race but this exposes all http/https endpoints defined when called. That's the more correct behavior for our default experience. Other http endpoints won't get exposed externally, which would blow up in ACA

Comment (timheuer) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs:14]: Will make the change, but not understanding how it affects the WithReference calls in this case.

Comment (davidfowl) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs]: It doesn't. It's about future calls to `WithHttpEndpoint`. If they get made before `WithExternalHttpEndpoints` it'll likely explode on ACA.

**Review by davidfowl — APPROVED** [review]

File: src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs (hunk 1 above)
Comment (timheuer) [reply src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.AppHost/Program.cs]: Sounds like another good case for an analyzer ;-)

**Review by eerhardt — APPROVED** [review]

**Comment by timheuer** [comment]
Assume we want to merge/backport this?

**Comment by davidfowl** [comment]
yes

**Comment by timheuer** [comment]
> yes

make it so all powerful one. 😜

**Pull request merged**